### PR TITLE
Edit reminder

### PR DESCRIPTION
**Pull request by 07nguyenpaul**
@stevekinney @martensonbj @brittanystoroz 

## Purpose

We need to make any reminder content editable by clicking on an edit button. This will allow the user to edit their title, date, or notes. 
We need to make a save button that allows the user to save their newly edited reminder.
https://github.com/turingschool-projects/1606-remember-3/issues/7 

## Approach

_How does this change address the problem?_
These changes allows the user to modify any of their reminders. 

Add controller/reminders/reminder.js file that contains new action function for edit and save.
Change reminders/reminder.hbs to have conditional if/else on title, date, and notes.  

### Learning

_Describe the research stage._
https://guides.emberjs.com/v1.10.0/getting-started/toggle-todo-editing-state/
https://guides.emberjs.com/v1.10.0/getting-started/accepting-edits/
http://jsfiddle.net/Kkzrb/8/

### Open Questions and Pre-Merge TODOs

- [ ] Use Github checklists. When solved, check the box and explain the answer.

### Test coverage 

_Briefly describe what tests you have written and why._

### Merge Dependencies and Related Work

_Link to merge dependency issues or PRs._

### CODE REVIEW

**Comment by brittanystoroz** [comment]
It looks like a lot of the code in this pull request is also in https://github.com/turingschool-projects/1606-remember-3/pull/8. This could cause merge conflicts down the road, require you to update the same code more than once, and cause reviewers to be giving the same feedback in different areas. Making the call for no one to review this right now until the other PRs are reviewed and merged.

**TLDR: Don't Review Right Now**

**Comment by Peter-Springer** [comment]
resolved conflicts...code review please

File: app/routes/reminders/new-reminder.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import Ember from 'ember';
+
+export default Ember.Route.extend({
+});
```
Comment (brittanystoroz) [inline app/routes/reminders/new-reminder.js]: Is this file doing something in particular?

File: app/templates/reminders/reminder.hbs [hunk 2]
```diff
@@ -1,8 +1,38 @@
...
+      </p>
+      <section class="button-container">
+        <button
+          class="edit-button"
+          {{action 'editReminder' on="click"}}>
+          edit
+        </button>
+        <button
+          class="save-button"
+          {{action 'saveEdits' on="click"}}>
```
Comment (brittanystoroz) [inline app/templates/reminders/reminder.hbs]: We probably don't want to display this save button unless they are actually editing the reminder. Or we should at least have it in a 'disabled' state when it's not being editing.

File: app/controllers/reminders/reminder.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+export default Ember.Controller.extend({
+  store: Ember.inject.service(),
+
+  isEditable: false,
+
+  actions: {
+    editReminder() {
+      this.set('isEditing', true);
+    },
+    saveEdits() {
```
Comment (brittanystoroz) [inline app/controllers/reminders/reminder.js]: Not something that will block your pull request getting merged, but `saveEdits` is kind of a weird name to me for some reason. I think in most codebases you're much more likely to see this type of method simply called `save` or `saveChanges`. 'Edits' feels a little more ambiguous.

File: app/routes/reminders/new-reminder.js (hunk 1 above)
Comment (07nguyenpaul) [inline app/routes/reminders/new-reminder.js]: Deleted file.
